src/terrain/path/tile_iterator.cpp

```diff
@@ -22,94 +22,85 @@
  *
  */
 
-#include <cstdint>
-#include <utility>
 #include "../terrain.hpp"
 #include "unit_path.hpp"
 
-namespace terrain{
-
-namespace path{
-
-static uint8_t offsets[26] = {0b00000000,
-                       0b00000001,
-                       0b00000010,
-                       0b00000100,
-                       0b00000101,
-                       0b00000110,
-                       0b00001000,
-                       0b00001001,
-                       0b00001010,
-                       0b00010000,
-                       0b00010001,
-                       0b00010010,
-                       0b00010100,
-                       0b00010110,
-                       0b00011000,
-                       0b00011001,
-                       0b00011010,
-                       0b00100000,
-                       0b00100001,
-                       0b00100010,
-                       0b00100100,
-                       0b00100101,
-                       0b00100110,
-                       0b00101000,
-                       0b00101001,
-                       0b00101010 };
-
-std::array<int8_t, 3> get_indexed_offsets(uint8_t index){
+#include <cstdint>
+#include <utility>
+
+namespace terrain {
+
+namespace path {
+
+static uint8_t offsets[26] = {
```

Comment:
Either make this number into a `#define`/add a comment or remove it and just leave the curly braces for the compiler to decide:

```cpp
// This
#define NUM_OFFSETS 26
static uint8_t offsets[NUM_OFFSETS] = /* */

// OR
/* We have 26 offsets because... */
static uint8_t offsets[26] = /* */

// OR
static uint8_t offsets[] = /* */
```